src/main.rs

```diff
@@ -52,7 +60,23 @@ fn check_file(arg: &str) {
             chname,
             node.kind()
         )
-        .with_source_code(NamedSource::new(arg, src.clone()));
+        .with_source_code(NamedSource::new(&filename, src.clone()));
```

Comment:
You have already cunstructed a `NamedSource` with identical arguments above. Can't this one be reused here?